ksiren/src/main/java/com/brightspace/ksiren/Entity.kt

```diff
@@ -18,12 +18,34 @@ package com.brightspace.ksiren
 class Entity(
 	val classes: List<String> = listOf(),
 	val properties: Map<String, String> = mapOf(),
-	val entities: List<Entity> = listOf(),
+	val entities: MutableList<Entity> = mutableListOf(),
```

Comment:
We should really avoid a `MutableList` here... It would be better to have the mutable list as a private backing property and expose `entities` as a `List<Entity>` to guarantee immutability.

Reply:
Oh interesting! I guess the idea would be that, to external code, the list of entities would be immutable, but `getSubEntitiesOfType()` would still be able to flesh out the entities behind the scenes?

Would making the entities variable private and forcing the use of `getSubEntitiesOfType()` be better, even? That way we don't just hand out embedded Entities that external applications can't really make any use of...